sentinel/src/transitions/accept_patient_reports.js

```diff
@@ -48,180 +48,228 @@ const getScheduledTasksByType = (registration, type) => {
-
-        return tasksUnderReview.filter(({group, type}) =>
-            hasGroupAndType(groupTypeCombosToClear, [group, type]));
-    }
+  if (!config.silence_for) {
+    // No range, all clearable tasks should be cleared
+    return tasksUnderReview.filter(task => typesToClear.includes(task.state));
+  } else {
+    // Clear all tasks that are members of a group that "exists" before the
+    // silenceUntil date. e.g., they have at least one task in their group
+    // whose due date is before silenceUntil.
+    const silenceUntil = reportedDateMoment.clone();
+    silenceUntil.add(date.getDuration(config.silence_for));
+
+    const allTasksBeforeSilenceUntil = tasksUnderReview.filter(
+      task => moment(task.due) <= silenceUntil
+    );
+    const groupTypeCombosToClear = uniqueGroupTypeCombos(
+      allTasksBeforeSilenceUntil
+    );
+
+    return tasksUnderReview.filter(({ group, type }) =>
+      hasGroupAndType(groupTypeCombosToClear, [group, type])
+    );
+  }
 };
 
 const getConfig = function(form) {
-    const fullConfig = config.get('patient_reports') || [];
-    return _.findWhere(fullConfig, { form: form });
+  const fullConfig = config.get('patient_reports') || [];
+  return _.findWhere(fullConfig, { form: form });
 };
 
 const _silenceReminders = (registration, report, config, callback) => {
-    var toClear = module.exports._findToClear(registration, report.reported_date, config);
-    if (!toClear.length) {
-        return callback();
-    }
+  var toClear = module.exports._findToClear(
+    registration,
+    report.reported_date,
+    config
+  );
+  if (!toClear.length) {
+    return callback();
+  }
 
-    toClear.forEach(task => {
-        utils.setTaskState(task, 'cleared');
-        task.cleared_by = report._id;
-    });
-    db.audit.saveDoc(registration, callback);
+  toClear.forEach(task => {
+    utils.setTaskState(task, 'cleared');
+    task.cleared_by = report._id;
+  });
+  db.audit.saveDoc(registration, callback);
 };
 
 const addRegistrationToDoc = (doc, registrations) => {
-    if (registrations.length) {
-        const latest = _.max(
-            registrations,
-            registration => moment(registration.reported_date)
-        );
-        doc.registration_id = latest._id;
-    }
+  if (registrations.length) {
+    const latest = _.max(registrations, registration =>
+      moment(registration.reported_date)
+    );
+    doc.registration_id = latest._id;
+  }
 };
 
-const silenceRegistrations = (
-            config,
-            doc,
-            registrations,
-            callback) => {
-    if (!config.silence_type) {
-        return callback(null, true);
-    }
-    async.forEach(
-        registrations,
-        function(registration, callback) {
-            if (doc._id === registration._id) {
-                // don't silence the registration you're processing
-                return callback();
+const addMessageUUIDToDoc = (doc, registrations, patientId) => {
```

Comment:
This function in general is quite complicated. It looks like a good first draft, but I think it needs to be workshopped more to reduce it's complexity, it's reliance on hardcoding, and introduce more code re-use and consistency. Tests are definitely also required here.

I haven't made individual notes or gone too deeply into reviewing the function, since I imagine removing the hard coding might change the implementation substantially, but let me know on Slack if you want me to write more detailed notes here.